### DISCUSSION TITLE
Unlock and copy buttons in "tab_view" mode

### DISCUSSION THREAD

**Discussion by juliop4s** [Onemind-Services-LLC/netbox-secrets#114] [General]
Hi =)

I noticed that when the plugin is configured to use the tab_view, the unlock and copy buttons are not directly available. They only appear on the Edit mode.

Is it possible to configure the plugin in a way that these action buttons are directly displayed also in tab mode?

Thank you!

![image](https://github.com/Onemind-Services-LLC/netbox-secrets/assets/62614723/fc960125-53d6-4e87-a93a-4bb1b91b38d1)

**Comment by mikedlr** [comment]
The difficulty with this seems to be that the [copy button is an external component called a sphinx-copybutton](https://github.com/search?q=org%3AOnemind-Services-LLC+copyButton&type=code) which comes from the [sphinx-copybutton Sphinx extension](https://github.com/executablebooks/sphinx-copybutton). 

In order to get this working someone would likely have to copy /"vendor" the module into NetBox, rename it and make  version that included code for decrypting the secret when its encrypted and a session is available. That would probably also have to ask the user to start a session if they didn't have a secrets session already. 

@juliop4s are you interested in doing anything about this? I wonder if it shouldn't be a bug or something rather than just a discussion. The way it works now is not nice when using Netbox in a shared working room.
